ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -0,0 +1,247 @@
+    const QString DEFAULT_STYLESHEET = "QProgressBar:horizontal {\
+            border: 1px solid white;\
+            border-radius: 7px;\
+            background: black;\
+            }\
+            QProgressBar::chunk:horizontal{\
+            border-radius: 7px;\
+            background: ";
+}
+
+RaceModeDashboardView::RaceModeDashboardView(BatteryPresenter& batteryPresenter,
+        BatteryFaultsPresenter& batteryFaultsPresenter,
+        DriverControlsPresenter& driverControlsPresenter,
+        KeyMotorPresenter& keyMotorPresenter,
+        LightsPresenter& lightsPresenter,
+        MpptPresenter& mpptPresenter,
+        MotorDetailsPresenter& motorDetailsPresenter,
+        MotorFaultsPresenter& motorFaultsPresenter,
+        I_RaceModeDashboardUI& ui)
+    : batteryPresenter_(batteryPresenter)
+    , batteryFaultsPresenter_(batteryFaultsPresenter)
+    , driverControlsPresenter_(driverControlsPresenter)
+    , keyMotorPresenter_(keyMotorPresenter)
+    , lightsPresenter_(lightsPresenter)
+    , mpptPresenter_(mpptPresenter)
+    , motorDetailsPresenter_(motorDetailsPresenter)
+    , motorFaultsPresenter_(motorFaultsPresenter)
+    , ui_(ui)
+{
+    connectBattery(batteryPresenter_);
+    connectBatteryFaults(batteryFaultsPresenter_);
+    connectDriverControls(driverControlsPresenter_);
+    connectKeyMotor(keyMotorPresenter_);
+    connectLights(lightsPresenter_);
+    connectMppt(mpptPresenter_);
+    connectMotorDetails(motorDetailsPresenter_);
+    connectMotorFaults(motorFaultsPresenter_);
+    //ui_.showMaximized();
+    ui_.show();
+}
+RaceModeDashboardView::~RaceModeDashboardView()
+{
+}
+
+void RaceModeDashboardView::connectBattery(BatteryPresenter& batteryPresenter)
+{
+    // TODO update to new battery data (depends on what should be shown in UI)
+    connect(&batteryPresenter, SIGNAL(aliveReceived(bool)),
+            this, SLOT(aliveReceived(bool)));
+    connect(&batteryPresenter, SIGNAL(prechargeStateReceived(QString)),
+            this, SLOT(prechargeStateReceived(QString)));
+    connect(&batteryPresenter, SIGNAL(packNetPowerReceived(double)),
+            this, SLOT(packNetPowerReceived(double)));
+}
+
+void RaceModeDashboardView::connectBatteryFaults(BatteryFaultsPresenter& batteryFaultsPresenter)
+{
+    connect(&batteryFaultsPresenter, SIGNAL(errorFlagsReceived(BatteryErrorFlags)),
+            this, SLOT(errorFlagsReceived(BatteryErrorFlags)));
+    connect(&batteryFaultsPresenter, SIGNAL(limitFlagsReceived(BatteryLimitFlags)),
+            this, SLOT(limitFlagsReceived(BatteryLimitFlags)));
+}
+
+void RaceModeDashboardView::connectDriverControls(DriverControlsPresenter& driverControlsPresenter)
+{
+    connect(&driverControlsPresenter, SIGNAL(resetReceived(bool)),
+            this, SLOT(resetReceived(bool)));
+}
+
+void RaceModeDashboardView::connectKeyMotor(KeyMotorPresenter& keyMotorPresenter)
+{
+    connect(&keyMotorPresenter, SIGNAL(motorSetCurrentReceived(double setCurrent)),
```

Comment:
Update the connects here to be 
`connect(&keyMotorPresenter, SIGNAL(motorActualSpeedReceived(double),  this, SLOT(motorActualSpeedReceived(double)));`

Having the variable name turns out to be wrong and causes the object not to react properly.